plugin/gstplayer.cpp

```diff
@@ -333,19 +333,25 @@ QtGstPlayer::setState(PipelineState state)
 	if (m_sink->property("itemInitialized").toBool()) {
 		switch(state) {
 		case PipelineState::Play:
-			QtConcurrent::run(gst_element_set_state, m_pipeline, GST_STATE_PLAYING);
+			QThreadPool::globalInstance()->start([=] {
+				gst_element_set_state(m_pipeline, GST_STATE_PLAYING);
+			});
```

Comment:
Setting the pipeline state may fail. Maybe we should actually handle the failure, or at least inform the user. However, we don't handle the failure now, and we probably don't need the `QFuture` to handle the failure. Thus, this is OK.